src/vehicle.cpp

```diff
@@ -4451,13 +4429,15 @@ void vehicle::consume_fuel( int load, const int t_seconds, bool skip_electric )
         double amnt_precise_j = static_cast<double>( fuel_pr.second ) * t_seconds;
         amnt_precise_j *= load / 1000.0 * ( 1.0 + st * st * 100.0 );
         auto inserted = fuel_used_last_turn.insert( { ft, 0.0f } );
-        inserted.first->second += amnt_precise_j;
         double remainder = fuel_remainder[ ft ];
         amnt_precise_j -= remainder;
 
         if( amnt_precise_j > 0.0f ) {
-            fuel_remainder[ ft ] = drain_energy( ft, amnt_precise_j ) - amnt_precise_j;
+            double actual_drain = drain_energy( ft, amnt_precise_j );
```

Comment:
Why are you messing with the functions that handle fuel consumption in a PR that is supposed to be about displaying fuel consumption?

If this function is buggy, open a separate issue and a PR to fix to it, don't do it in this one please.